File: member-service/src/main/java/com/hedvig/memberservice/events/OnboardingStartedWithSSNEvent.java
package com.hedvig.memberservice.events;

import lombok.EqualsAndHashCode;
import org.axonframework.serialization.Revision;

//This event is deprecated don't use it for any new stuff
@EqualsAndHashCode(callSuper = true)
@Deprecated
@Revision("1.0")
public class OnboardingStartedWithSSNEvent extends SSNUpdatedEvent {

  public OnboardingStartedWithSSNEvent(Long memberId, String ssn, Nationality nationality) {
    super(memberId, ssn, nationality);
  }

    @Override
    public String getSsn() {
        return super.getSsn();
    }
}

File: member-service/src/main/java/com/hedvig/memberservice/events/OnboardingStartedWithSSNEvent.kt
package com.hedvig.memberservice.events

import org.axonframework.serialization.Revision
import org.axonframework.serialization.upcasting.event.EventUpcaster

@Deprecated("This event is deprecated don't use it for any new stuff")
@Revision("1.0")
class OnboardingStartedWithSSNEvent(
    memberId: Long,
    ssn: String,
    nationality: Nationality
) : SSNUpdatedEvent(memberId, ssn, nationality) {

    companion object {
        val upcasters: List<EventUpcaster> = listOf(
            NationalityFromSsnUpcaster("com.hedvig.memberservice.events.OnboardingStartedWithSSNEvent")
        )
    }
}
